pulser-core/pulser/register/base_register.py

```diff
@@ -202,10 +206,28 @@ def _validate_layout(
 
     @abstractmethod
     def _to_dict(self) -> dict[str, Any]:
-        qs = dict(zip(self._ids, map(np.ndarray.tolist, self._coords)))
-        if self._layout_info is not None:
-            return obj_to_dict(self, qs, **(self._layout_info._asdict()))
-        return obj_to_dict(self, qs)
+        cls_dict = obj_to_dict(
+            None,
+            _build=False,
+            _name=self.__class__.__name__,
+            _module=self.__class__.__module__,
+        )
+
+        kwargs = (
+            {} if self._layout_info is None else self._layout_info._asdict()
+        )
+
+        return obj_to_dict(
+            self,
+            cls_dict,
+            [np.ndarray.tolist(qubit_coords) for qubit_coords in self._coords],
+            False,
+            None,
+            self._ids,
+            **kwargs,
+            _submodule=self.__class__.__name__,
+            _name="from_coordinates",
+        )
```

Comment:
Great! Can you maybe add just a comment explaining why we're using `from_coordinates`? I feel like this complexity might be confusing without an explanation